### PR TITLE
Sanitize URL when it is copied into clipboard by JS.

### PR DESCRIPTION
**Pull request by boocmp**
Resolves https://github.com/brave/brave-browser/issues/33037

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See in the issue
Test page https://boocmp.github.io/clipboard/

### CODE REVIEW

File: browser/url_sanitizer/url_sanitizer_browsertest.cc [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+#include "net/test/embedded_test_server/embedded_test_server.h"
+#include "ui/base/clipboard/clipboard.h"
+#include "ui/base/clipboard/scoped_clipboard_writer.h"
+
+namespace {
+
+constexpr const char kYoutubeRules[] = R"json(
+    [{
+        "include": [
+            "*://youtu.be/*?*",
```
Comment (fmarier) [inline browser/url_sanitizer/url_sanitizer_browsertest.cc:29]: This can be just `*://youtu.be/*`.

File: browser/brave_browser_features.cc [hunk 2]
```diff
@@ -24,6 +24,10 @@ BASE_FEATURE(kBraveCopyCleanLinkByDefault,
 #endif
 );
 
+BASE_FEATURE(kBraveCopyCleanLinkFromJs,
+             "brave-copy-clean-link-from-js",
```
Comment (iefremov) [inline browser/brave_browser_features.cc:28]: can we name it `BraveCopyCleanLinkFromJs` for ease of search

File: test/BUILD.gn [hunk 3]
```diff
@@ -1119,6 +1119,7 @@ test("brave_browser_tests") {
       "//brave/browser/brave_resources_browsertest.cc",
       "//brave/browser/ssl/certificate_transparency_browsertest.cc",
       "//brave/browser/ui/views/toolbar/wallet_button_notification_source_browsertest.cc",
+      "//brave/browser/url_sanitizer/url_sanitizer_browsertest.cc",
```
Comment (iefremov) [inline test/BUILD.gn:1122]: put it into a sepratate browser_tests target in b/b/url_sanitizer/BUILD.gn
this old way is kinda deprecated

File: browser/brave_content_browser_client.cc [hunk 4]
```diff
@@ -1387,3 +1390,21 @@ blink::UserAgentMetadata BraveContentBrowserClient::GetUserAgentMetadata() {
...
+
+GURL BraveContentBrowserClient::SanitizeURL(
+    content::RenderFrameHost* render_frame_host,
+    const GURL& url) {
+  if (!base::FeatureList::IsEnabled(features::kBraveCopyCleanLinkFromJs)) {
+    return url;
+  }
+  DCHECK(render_frame_host);
+  DCHECK(render_frame_host->GetBrowserContext());
+  auto* url_sanitizer_service =
```
Comment (iefremov) [inline browser/brave_content_browser_client.cc:1402]: maybe CHECK it as well? is it always non-null in tor/incognito ?

**Review by fmarier — APPROVED** [review]

**Review by iefremov — APPROVED** [review]

**Pull request merged**